File: 2021-Shenzhen-FinTechathon3/Taishang-Credential-With-Interactive-Badges/Project/fisco_bcos_python_sdk_updated/client/precompile/crud/condition.py
'''
  FISCO BCOS/Python-SDK is a python client for FISCO BCOS2.0 (https://github.com/FISCO-BCOS/)
  FISCO BCOS/Python-SDK is free software: you can redistribute it and/or modify it under the
  terms of the MIT License as published by the Free Software Foundation. This project is
  distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even
  the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. Thanks for
  authors and contributors of eth-abi, eth-account, eth-hash，eth-keys, eth-typing, eth-utils,
  rlp, eth-rlp , hexbytes ... and relative projects
  @file: condition.py
  @function:
  @author: yujiechen
  @date: 2019-07
'''


class ConditionOp:
    """
    define operations for condition
    """
    _EQ_ = "eq"
    _NE_ = "ne"
    _GT_ = "gt"
    _GE_ = "ge"
    _LT_ = "lt"
    _LE_ = "le"
    _LIMIT_ = "limit"


class Condition:
    """
    implementation of Condition
    """

    def __init__(self):
        self.conditions = {}

    def update_condition(self, op, key, value):
        """
        update condition with the given (op, key, value)
        """
        if self.conditions.get(key) is not None:
            if self.conditions.get(key).get(op) is not None:
                self.conditions[key].get(op).put(value)
            else:
                self.conditions[key][op] = value
            return
        op_dict = {}
        op_dict[op] = value
        self.conditions[key] = op_dict

    def eq(self, key, value):
        self.update_condition(ConditionOp._EQ_, key, value)

    def ne(self, key, value):
        self.update_condition(ConditionOp._NE_, key, value)

    def gt(self, key, value):
        self.update_condition(ConditionOp._GT_, key, value)

    def ge(self, key, value):
        self.update_condition(ConditionOp._GE_, key, value)

    def lt(self, key, value):
        self.update_condition(ConditionOp._LT_, key, value)

    def le(self, key, value):
        self.update_condition(ConditionOp._LE_, key, value)

    def limit(self, offset, count):
        if offset < 0:
            offset = 0
        if count < 0:
            count = 0
        op_dict = {}
        op_dict[ConditionOp._LIMIT_] = str(offset) + "," + str(count)
        self.conditions["limit"] = op_dict

    def get_conditions(self):
        return self.conditions

    def set_conditions(self, conditions):
        self.conditions = conditions

File: 2024-shenzhen-FinTechathon/AIGCCM/go-sdk/precompiled/crud/condition.go
package crud

import "strconv"

type EnumOP string

const (
	EQ    EnumOP = "eq"
	NE    EnumOP = "ne"
	GT    EnumOP = "gt"
	GE    EnumOP = "ge"
	LT    EnumOP = "lt"
	LE    EnumOP = "le"
	Limit EnumOP = "limit"
)

type Condition struct {
	conditions map[string]map[EnumOP]string
}

func NewCondition() *Condition {
	return &Condition{conditions: make(map[string]map[EnumOP]string)}
}

func (c *Condition) EQ(key string, value string) {
	newMap := make(map[EnumOP]string)
	newMap[EQ] = value
	c.conditions[key] = newMap
}

func (c *Condition) NE(key string, value string) {
	newMap := make(map[EnumOP]string)
	newMap[NE] = value
	c.conditions[key] = newMap
}

func (c *Condition) GT(key string, value string) {
	newMap := make(map[EnumOP]string)
	newMap[GT] = value
	c.conditions[key] = newMap
}

func (c *Condition) GE(key string, value string) {
	newMap := make(map[EnumOP]string)
	newMap[GE] = value
	c.conditions[key] = newMap
}

func (c *Condition) LT(key string, value string) {
	newMap := make(map[EnumOP]string)
	newMap[LT] = value
	c.conditions[key] = newMap
}

func (c *Condition) LE(key string, value string) {
	newMap := make(map[EnumOP]string)
	newMap[LE] = value
	c.conditions[key] = newMap
}

func (c *Condition) Limit(count int) {
	c.limit(0, count)
}

func (c *Condition) limit(offset int, count int) {
	newMap := make(map[EnumOP]string)
	if offset < 0 {
		offset = 0
	}
	if count < 0 {
		count = 0
	}
	newMap[Limit] = strconv.Itoa(offset) + "," + strconv.Itoa(count)
	c.conditions["limit"] = newMap
}

func (c *Condition) GetConditions() map[string]map[EnumOP]string {
	return c.conditions
}

func (c *Condition) SetConditions(newMap map[string]map[EnumOP]string) {
	c.conditions = newMap
}
